composer/datasets/ade20k.py

```diff
@@ -0,0 +1,336 @@
+
+        if contrast_mode == 0 and np.random.randint(2):
+            contrast_factor = np.random.uniform(1 - self.contrast, 1 + self.contrast)
+            image = TF.adjust_contrast(image, contrast_factor)  # type: ignore
+
+        return image
+
+
+class ADE20k(Dataset):
+    """PyTorch Dataset for ADE20k.
+
+    Args:
+        datadir (str): the path to the ADE20k folder.
+        split (str): the dataset split to use, either 'train', 'val', or 'test'.
+        both_transforms (torch.nn.Module): transformations to apply to the image and target simultaneously.
+        image_transforms (torch.nn.Module): transformations to apply to the image only.
+        target_transforms (torch.nn.Module): transformations to apply to the target only.
+    """
+
+    def __init__(self, datadir: str, split: str, both_transforms: torch.nn.Module, image_transforms: torch.nn.Module,
+                 target_transforms: torch.nn.Module):
+        super().__init__()
+        self.datadir = datadir
+        self.split = split
+        self.both_transforms = both_transforms
+        self.image_transforms = image_transforms
+        self.target_transforms = target_transforms
+
+        self.image_files = os.listdir(os.path.join(self.datadir, 'images', self.split))
+
+        # Filter for ADE files
+        self.image_files = [f for f in self.image_files if f[:3] == 'ADE']
+
+        # Remove grayscale samples
+        if self.split == 'train':
+            corrupted_samples = ['00003020', '00001701', '00013508', '00008455']
+            for sample in corrupted_samples:
+                self.image_files.remove(f'ADE_train_{sample}.jpg')
+
+    def __getitem__(self, index):
+        # Load image
+        image_file = self.image_files[index]
+        image_path = os.path.join(self.datadir, 'images', self.split, image_file)
+        image = Image.open(image_path)
+
+        # Load annotation target if using either train or val splits
+        if self.split in ['train', 'val']:
+            target_path = os.path.join(self.datadir, 'annotations', self.split, image_file.split('.')[0] + '.png')
+            target = Image.open(target_path)
+
+            if self.both_transforms:
+                image, target = self.both_transforms((image, target))
+
+            if self.target_transforms:
+                target = self.target_transforms(target)
+
+        if self.image_transforms:
+            image = self.image_transforms(image)
+
+        if self.split in ['train', 'val']:
+            return image, target  # type: ignore
+        else:
+            return image
+
+    def __len__(self):
+        return len(self.image_files)
+
+
+@dataclass
+class ADE20kDatasetHparams(DatasetHparams):
+    """ Defines an instance of the ADE20k dataset for semantic segmentation.
+
+    Args:
+        split (str): the dataset split to use, either train, val, or test.
+        base_size (int): initial size of the image and target before other augmentations.
```

Comment:
Can you include the defaults in the docstrings?

Reply:
Defaults added!